packages/cli/generators/discover/index.js

```diff
@@ -291,14 +290,35 @@ module.exports = class DiscoveryGenerator extends ArtifactGenerator {
     // eslint-disable-next-line @typescript-eslint/prefer-for-of
     for (let i = 0; i < this.discoveringModels.length; i++) {
       const modelInfo = this.discoveringModels[i];
+      // passing connector specific options from the cli through connectorDiscoveryOptions
+      let discoveryOptions = {};
+      if (this.options.connectorDiscoveryOptions) {
+        discoveryOptions = JSON.parse(this.options.connectorDiscoveryOptions);
+      }
       debug(`Discovering: ${modelInfo.name}...`);
+      // when options are passed to discover command as --config, the treatTINYINT1AsTinyInt comes as a string
+      if (typeof discoveryOptions.treatCHAR1AsString === 'string') {
```

Comment:
These are very specific to mysql. Generator code need to remain generic for all types of connector.. I don't think we should add these here. We can simply pass on the config to connector and handle parsing of the options at individual connector level.